protos/flyteidl/plugins/sagemaker/trainingjob.proto

```diff
@@ -0,0 +1,45 @@
+syntax = "proto3";
+
+package flyteidl.plugins.sagemaker;
+
+option go_package = "github.com/lyft/flyteidl/gen/pb-go/flyteidl/plugins";
+
+
+message AlgorithmSpecification {
+    string training_image = 1;  // TODO: will this change output? should it be an input instead of a part of the task spec?
+    string training_input_mode = 2;
+    string algorithm_name = 3;   // TODO: will this change output? should it be an input instead of a part of the task spec?
+
+    message MetricDefinition {
+        string name = 1;
+        string regex = 2;
+    }
+    repeated MetricDefinition metric_definitions = 4;
+}
+
+message ResourceConfig {
+    string instance_type = 1;
+    int64 instance_count = 2;
+    int64 volume_size_in_gb = 3;
+    string volume_kms_key_id = 4;
+}
+
+message StoppingCondition {
+    int64 max_runtime_in_seconds = 1;
+    int64 max_wait_time_in_seconds = 2;
+}
+
+message VpcConfig {
+    repeated string security_group_ids = 1;
+    repeated string subnets = 2;
+}
+
+message TrainingJob {
+    string region = 1;
+    string role_arn = 2;
```

Comment:
I think this should be the same role that we are using for the workflow?

Reply:
Right. That means we do not need to define this as a part of the message?